### PR TITLE
TEST/UCM: Lock memory in mmap_ptrs test to ensure consistent results.

### LINKED ISSUE #1994: failure on malloc_hook_cplusplus.mmmap_ptrs
```
12:28:35 [----------] 1 test from malloc_hook_cplusplus, where TypeParam = 
12:28:35 [ RUN      ] malloc_hook_cplusplus.mmmap_ptrs
12:28:48 [     INFO ] Reduced heap free space to minimum: 9996816 after allocating 185980000 bytes
12:28:48 [     INFO ] With 11 large blocks: allocated 100000 buffers of 524288 bytes in 0.401033 sec
12:28:51 [     INFO ] With 8203 large blocks: allocated 100000 buffers of 524288 bytes in 0.805148 sec
12:28:51 /scrap/jenkins/scrap/workspace/hpc-ucx-pr-2/label/hpc-test-node2/worker/1/contrib/../test/gtest/ucm/malloc_hook.cc:545: Failure
12:28:51 The difference between alloc_time and alloc_time_with_ptrs is 0.40411562153847563, which exceeds eps, where
12:28:51 alloc_time evaluates to 0.40103278346155252,
12:28:51 alloc_time_with_ptrs evaluates to 0.80514840500002816, and
12:28:51 eps evaluates to 0.30077458759616438.
12:28:52 
[  FAILED  ] malloc_hook_cplusplus.mmmap_ptrs, where TypeParam =  and GetParam() =  (16521 ms)
12:28:52 [----------] 1 test from malloc_hook_cplusplus (16521 ms total)
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5147/label=hpc-test-node2,worker=1/console

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #1994

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5332/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5348/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5355/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5358/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov pls take a look

File: test/gtest/ucm/malloc_hook.cc [hunk 1]
```diff
@@ -506,17 +506,20 @@ UCS_TEST_F(malloc_hook_cplusplus, mmmap_ptrs) {
     set();
 
     const size_t   size    = ucm_dlmallopt_get(M_MMAP_THRESHOLD) * 2;
-    const size_t   max_mem = ucs_min(ucs_get_phys_mem_size() / 4, 4 * UCS_GBYTE);
+    const size_t   max_mem = ucs_min(ucs_get_phys_mem_size() / 8, 4 * UCS_GBYTE);
```
Comment (shamisp) [inline test/gtest/ucm/malloc_hook.cc]: and why it changed to a magic value of 8 ?

File: test/gtest/ucm/malloc_hook.cc [hunk 2]
```diff
@@ -506,17 +506,20 @@ UCS_TEST_F(malloc_hook_cplusplus, mmmap_ptrs) {
...
     std::vector<std::string> ptrs;
 
     size_t large_blocks = 0;
 
-    /* Allocate until we get MMAP event */
+    /* Allocate until we get MMAP event
+     * Lock memory to avoid going to swap and ensure consistet test results.
+     */
     while (m_mapped_size == 0) {
         std::string str(size, 'r');
```
Comment (evgeny-leksikov) [inline test/gtest/ucm/malloc_hook.cc]: it's better to use std::vector here because std::string allocates at least +1 for null-terminator.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5378/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5380/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5382/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5391/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5393/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov @shamisp fixed

**Pull request merged**